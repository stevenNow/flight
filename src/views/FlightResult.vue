<template>
  <div class="center" style="margin-top:200px">
    <a-row>
      <a-col span="12"><h1>Search results</h1></a-col>
      <a-col offset="8" span="4"><a-button @click="backToSearch">Back</a-button></a-col>
    </a-row>
    <a-spin size="large" v-if="searching" class="spinner"></a-spin>
  <div v-for="(item, index) in results" :key="index" >
    <a-descriptions :title="'Flight ' + index + ' for $' + item.fare" class="searchListItem">
      <a-descriptions-item label="Origin">
        {{ item.origin }}
      </a-descriptions-item>
      <a-descriptions-item label="Departure">
        {{ item.departure }}
      </a-descriptions-item>
      <a-descriptions-item label="Destination">
        {{ item.destination }}
      </a-descriptions-item>
      <a-descriptions-item label="Arrival">
        {{ item.arrival }}
      </a-descriptions-item>
      <a-descriptions-item label="Cabin Type">
        {{ item.cabinType }}
      </a-descriptions-item>
    </a-descriptions>
  </div>
  </div>
</template>

<script>

export default {
  name: "FlightResult",
  components: {},
  created() {
    setTimeout(() => {
      const data = require("../data/flightResults.json");
      this.results = data;
      this.searching = false;
    }, 500)
  },
  data() {
    return {
      results: null,
      searching: true
    };
  },
  methods: {
    backToSearch() {
      this.$router.push({path: "/"})
    }
  }
};
</script>
<style>
.spinner {
  margin-left: 50%;
  margin-top: 100px;
}
.searchListItem {
  padding: 25px;
}
.searchListItem:hover {
  cursor: pointer;
  box-shadow: 0 0rem 0.5rem rgba(0, 0, 0, 0.15);
}
</style>