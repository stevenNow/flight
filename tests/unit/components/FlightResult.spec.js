import Vuex from "vuex";
import { createLocalVue, mount } from "@vue/test-utils";
import Antd from "ant-design-vue";
import FlightResult from "../../../src/views/FlightResult";

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Antd);
jest.useFakeTimers();

describe("FlightResult Test", () => {

    it("Tests FlightResult", async () => {
        const wrapper = mount(FlightResult, {
            localVue,
            mocks: {
                $router: {
                    push: jest.fn(),
                },
            },
        });
        await wrapper.vm.$nextTick();
        expect(wrapper).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();
        expect(wrapper.vm.searching).toBeTruthy();
        expect(wrapper.vm.results).toBeNull();
        jest.advanceTimersByTime(501);
        expect(wrapper.vm.searching).toBeFalsy();
        expect(wrapper.vm.results).toStrictEqual(
            [
                {
                    "origin": "LAX",
                    "destination": "JFK",
                    "departure": "02-22-2022T07:35:00",
                    "arrival": "02-22-2022T12:50:00",
                    "cabinType": "economy",
                    "fare": "995.00"
                },
                {
                    "origin": "LAX",
                    "destination": "JFK",
                    "departure": "02-22-2022T07:50:00",
                    "arrival": "02-22-2022T13:20:00",
                    "cabinType": "economy",
                    "fare": "990.00"
                },
                {
                    "origin": "LAX",
                    "destination": "JFK",
                    "departure": "02-22-2022T07:25:00",
                    "arrival": "02-22-2022T12:30:00",
                    "cabinType": "economy",
                    "fare": "1025.00"
                }
            ]
        )

    });
});