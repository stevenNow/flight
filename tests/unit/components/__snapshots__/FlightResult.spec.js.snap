// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightResult Test Tests FlightResult 1`] = `
<div
  class="center"
  style="margin-top: 200px;"
>
  <div
    class="ant-row"
  >
    <div
      class="ant-col ant-col-12"
    >
      <h1>
        Search results
      </h1>
    </div>
     
    <div
      class="ant-col ant-col-4 ant-col-offset-8"
    >
      <button
        class="ant-btn"
        type="button"
      >
        <i
          aria-label="icon: arrow-left"
          class="anticon anticon-arrow-left"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="arrow-left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 0 0 0 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </i>
        <span>
          Back
        </span>
      </button>
    </div>
  </div>
   
  <div
    class="spinner ant-spin ant-spin-lg ant-spin-spinning"
  >
    <span
      class="ant-spin-dot ant-spin-dot-spin"
    >
      <i
        class="ant-spin-dot-item"
      />
      <i
        class="ant-spin-dot-item"
      />
      <i
        class="ant-spin-dot-item"
      />
      <i
        class="ant-spin-dot-item"
      />
    </span>
  </div>
   
</div>
`;
